Changing "Bibliography" to "References"

tjiagoM opened the pull request:
Hi,

While I was writing my thesis I've noticed that using "Bibliography" at the end is not technically correct, and that "References" should be used instead. To be more specific, "References" seems to be used when it only contains sources cited in the main text, while "Bibliography" is used when it also includes works that were not cited in the main text. These are some links I've googled with this info after discussing with some colleagues: 
https://studenthelp.secure.griffith.edu.au/app/answers/detail/a_id/1676/~/what-is-the-difference-between-a-reference-list-and-a-bibliography%3F
https://www.nottingham.ac.uk/studentservices/documents/whatarebibliographiesandreferences.pdf


Given that the current "Bibliography" is automatically generated when something is cited, I believe the default name should be "References" instead. If one wants to include something which is not cited, they need to manually change things in order to do that, which I believe is another reason to have "References" by default.

Happy to discuss this further, but here is the PR to start things already :)

urbas commented:
Thanks @tjiagoM!

I did a cursory look at [the list of Computer Lab's technical reports](https://www.cl.cam.ac.uk/techreports/). It looks like both "Bibliography" and "References" are well represented. So that's at least a good sign it's not a barrier to submission.

However, given that Bibliography seems to be broader in scope, I'd say it's safer to leave it as is. For example, authors can decide to use `\nocite{xyz}` (assuming they have a good reason to do so).

So, all in all, I'd vote to leave it as is. This is thankfully not hard-coded in the class, so people can always change from "Bibliography" to "References" in their own clones. Also, the fewer changes we make to `thesis.tex` (which is where people will make most of their changes) the easier it will be for people to adopt class fixes from this repo.

I hope this sounds sensible.

tjiagoM commented:
Hi @urbas !

Thanks a lot for your comments! :) I understand your reasoning, but I still think otherwise, so I will try to explain why.

The main issue in my opinion is that the default behaviour introduces a naming that is incorrect in most cases, if not all of them, given how people tend to work/write in our department with latex.  I might be wrong here, but I believe a lot of recent thesis have "Bibliography" just because it was the name that came by default in the template, and never used a `\nocite{}` or equivalent. For example, the theses I had in my folder to get ideas on thesis structure don't seem to have included any extra work other than that the ones cited in-line, and still used "Bibliography".

Another thing is that I didn't know about this difference between References/Bibliography until a colleague of mine told me, and I could bet some money that most of people just don't care or don't know; that's why I think having "References" by default is the safest one in our department.

In the meantime I also found that the template for the Department of Engineering uses "References" by default, a department which I'd say uses latex in a similar way to us :): https://www.overleaf.com/latex/templates/phd-thesis-template-for-cambridge-university-engineering-department-cued-latex-xelatex-and-lualatex-support-v2-dot-1/kgfqybfnqkdf
(I was curious to know how they did it and it's by redefining the `\bibname` like I'm proposing in this PR here: https://github.com/kks32/phd-thesis-template/blob/master/Preamble/preamble.tex)


With regards to Bibliography being broader in scope, I don't think that's exactly the case - they are just distinct things. If you just put references to things you cite, then technically i's not a Bibliography. 


In my opinion these two reasons justify well the need to change the default behaviour. If you still don't want to do it, I suggest that at least we should leave a comment with a quick explanation on this: for instance, instructions on how people could change if they want, rather than trying to google it. As "explicit is better than implicit", I guess it could be a nice thing to have a comment on that so people who don't know the differences know about it. Actually, I think this could still be a good thing to have even if the default behaviour is changed.

urbas commented:
Nice, can't argue with that! Okay, so let's get these bibliographies renamed to references then. :+1:

urbas commented on Samples/clean/thesis.tex:
```diff
@@ -83,9 +83,10 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% Bibliography
```
Maybe rename this comment too (and the other ones).

urbas reviewed (commented):
These is also a mention of bibliography in the `Samples/clean/01Preamble.tex` and `README.md`, could you change those too please?

tjiagoM commented:
@urbas just pushed a few updates as required, let me know what you think.

I also added a comment to recommend explicitly to change to "Bibliography" in cases like \nocite. I think it's small enough to be useful as an explicit reminder.

I also changed the .gitignore because somehow I had that .synctex.gz generated and I didn't want to include it :p I don't know whether you want to include a broader gitignore for latex like this one though: https://github.com/github/gitignore/blob/master/TeX.gitignore

urbas commented:
Looks great! Thanks @tjiagoM :+1: 

Regarding `.gitignore`: I'd vote to keep it at a minimum actually. Tbh, I should have removed that `.idea` entry and checked whether all of the ignores are actually still needed.
